sources/api/netdog/src/cli/generate_net_config.rs

```diff
@@ -40,7 +43,10 @@ pub(crate) fn run() -> Result<()> {
     let primary_interface = net_config
         .primary_interface()
         .context(error::GetPrimaryInterfaceSnafu)?;
-    write_primary_interface(primary_interface)?;
+    // Remove existing primary interface files since the current primary may have changed or may
+    // now be using a MAC address (via an override net.toml/reboot)
+    rm_old_primary_interface()?;
```

Comment:
nit: I'd prefer `remove` over `rm` since it's not a shell script 😀 
```suggestion
    remove_old_primary_interface()?;
```